cmd/api/src/auth/model.go

```diff
@@ -85,50 +88,85 @@ func (s idResolver) GetIdentity(ctx Context) (SimpleIdentity, error) {
 	}
 }
 
+type AuditLogger interface {
+	AppendAuditLog(ctx context.Context, entry model.AuditEntry) error
+}
+
 type Authorizer interface {
+	HasPermission(ctx Context, requiredPermission model.Permission, grantedPermissions model.Permissions) bool
 	AllowsPermission(ctx Context, requiredPermission model.Permission) bool
 	AllowsAllPermissions(ctx Context, requiredPermissions model.Permissions) bool
 	AllowsAtLeastOnePermission(ctx Context, requiredPermissions model.Permissions) bool
+	AuditLogUnauthorizedAccess(request *http.Request)
 }
 
-type authorizer struct{}
+type authorizer struct {
+	auditLogger         AuditLogger
+	permissionsResolver func(ctx Context) model.Permissions
```

Comment:
This turned out to not be necessary and needs to be removed, will force push update